wallet.go

```diff
@@ -1122,6 +1124,102 @@ func (w *Wallet) DumpWIFPrivateKey(addr btcutil.Address) (string, error) {
 	return wif.String(), nil
 }
 
+// ImportAddress imports a watch-only address to the wallet.
+func (w *Wallet) ImportAddress(addr btcutil.Address, bs *waddrmgr.BlockStamp,
+	rescan bool) (string, error) {
+
+	// The starting block for the key is the genesis block unless otherwise
+	// specified.
+	if bs == nil {
+		bs = &waddrmgr.BlockStamp{
+			Hash:   *activeNet.Params.GenesisHash,
+			Height: 0,
+		}
+	}
+
+	// Check whether addr is P2SH or P2PKH and import accordingly
```

Comment:
Rather than doing this, I'd suggest just doing a type switch on the address and moving this entire logic to the manager's `ImportAddress` function which should work with any address type.  The necessary decoding has already been done by the time it's an address and base58 decoding is quite slow.

Also, there needs to be a default case which returns an unsupported address error.

Something like:

```Go
switch addr := addr.(type) {
case *btcutil.AddressPubKeyHash:
	// Handle p2pkh addresses...

case *btcutil.AddressPubKey:
	// Handle p2pk addresses...

case *btcutil.AddressScriptHash:
	// Handle p2sh addresses...

default:
	// Unsupported address type error....
}
```